pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -199,6 +308,8 @@ func setDefaultDeploymentSpec(spec *appsv1.DeploymentSpec) {
 		progressDeadlineSeconds := int32(600)
 		spec.ProgressDeadlineSeconds = &progressDeadlineSeconds
 	}
+
+	spec.Template.Spec.ServiceAccountName = constants.KserveServiceAccountName
```

Comment:
without this oauth proxy does not work and uses the default service account in the namespace. Example : 
```
requestauth: 10.128.6.55:54444 tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:rawroutes:default" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope
``` 
we have the same pattern for modelmesh as well I'm not sure I'm understanding the issue

Reply:
These are the docs that guide users to customizing the ServiceAccount of the pod, to fit their needs: https://kserve.github.io/website/latest/modelserving/storage/s3/s3/#attach-secret-to-a-service-account

Notice that you can specify `spec.predictor.serviceAccountName`. So, it is user customizable. You either are breaking that feature, or that feature can break oauth-proxy.

This can be relevant for users using custom serving runtimes. Indeed, I'm planning on using it for restoring E2Es to a working state.